model/model.go

```diff
@@ -401,12 +401,21 @@ func (v *ViewCheckOption) String() string {
 
 // ViewInfo provides meta data describing a DB view.
 type ViewInfo struct {
-	Algorithm   ViewAlgorithm      `json:"view_algorithm"`
-	Definer     *auth.UserIdentity `json:"view_definer"`
-	Security    ViewSecurity       `json:"view_security"`
-	SelectStmt  string             `json:"view_select"`
-	CheckOption ViewCheckOption    `json:"view_checkoption"`
-	Cols        []CIStr            `json:"view_cols"`
+	Algorithm ViewAlgorithm      `json:"view_algorithm"`
+	Definer   *auth.UserIdentity `json:"view_definer"`
+	Security  ViewSecurity       `json:"view_security"`
+	//In order to support ansi_quotes sqlmode, we need to store SelectStmt with different quotes.
+	SelectStmt  []string        `json:"view_select"`
```

Comment:
Can we create a new field called SelectStmtANSIQuote instead of replace this into a slice?

This breaks backward compatible, since if we use a old version of TiDB to open the new database, it will throw:

```
2019/03/22 15:51:03.950 tidb.go:79: [error] [ddl] init domain failed json: cannot unmarshal array into Go struct field ViewInfo.view_select of type string
github.com/pingcap/errors.AddStack
        /home/ejq/go/pkg/mod/github.com/pingcap/errors@v0.11.1/errors.go:174
github.com/pingcap/errors.Trace
        /home/ejq/go/pkg/mod/github.com/pingcap/errors@v0.11.1/juju_adaptor.go:15
github.com/pingcap/tidb/meta.(*Meta).ListTables
        /home/ejq/go/src/github.com/pingcap/tidb/meta/meta.go:398
github.com/pingcap/tidb/domain.(*Domain).fetchSchemasWithTables
        /home/ejq/go/src/github.com/pingcap/tidb/domain/domain.go:177
runtime.goexit
        /usr/lib/go/src/runtime/asm_amd64.s:1337
```

When starting up.

Another suggestion is that since show is a low frequency command we can just store the select statement quoted by back quotes, and rebuild it with double quotes when we need to show the statement.

Reply:
It's not only used b `SHOW CREATE` statement, `SELECT` statement use this too.

Reply:
So maybe we should use two fields instead of replace it by a slice

Reply:
Or maybe we can force parser to use back quotes when parsing views in select.